File: Assets/Bundles/Server/Resources/TS/Services/Global/Chat/Commands/LagCommand.lua
-- Compiled with unity-ts v2.1.0-75
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local LagCommand
do
	local super = ChatCommand
	LagCommand = setmetatable({}, {
		__tostring = function()
			return "LagCommand"
		end,
		__index = super,
	})
	LagCommand.__index = LagCommand
	function LagCommand.new(...)
		local self = setmetatable({}, LagCommand)
		return self:constructor(...) or self
	end
	function LagCommand:constructor()
		super.constructor(self, "lag")
	end
	function LagCommand:Execute(player, args)
		local transportManager = GameObject:Find("Network"):GetComponent("TransportManager")
		if transportManager.LatencySimulator:GetEnabled() and (#args == 0 or string.lower(args[1]) == "off") then
			transportManager.LatencySimulator:SetEnabled(false)
			player:SendMessage("Disabled lag sim.")
			return nil
		end
		local latency = 90
		if #args > 0 then
			local num = tonumber(args[1])
			if num ~= nil then
				latency = num
			end
		end
		player:SendMessage("Enabled lag sim of " .. (tostring(latency) .. "ms"))
		transportManager.LatencySimulator:SetLatency(latency)
		transportManager.LatencySimulator:SetEnabled(true)
	end
end
return {
	LagCommand = LagCommand,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Server/Services/Global/Chat/Commands/LagCommand.ts
import { Player } from "Shared/Player/Player";
import { ChatCommand } from "../../../../Commands/ChatCommand";

export class LagCommand extends ChatCommand {
	constructor() {
		super("lag");
	}

	public Execute(player: Player, args: string[]): void {
		const transportManager = GameObject.Find("Network").GetComponent<TransportManager>();

		if (transportManager.LatencySimulator.GetEnabled() && (args.size() === 0 || args[0].lower() === "off")) {
			transportManager.LatencySimulator.SetEnabled(false);
			player.SendMessage("Disabled lag sim.");
			return;
		}

		let latency = 90;
		if (args.size() > 0) {
			const num = tonumber(args[0]);
			if (num !== undefined) {
				latency = num;
			}
		}
		player.SendMessage(`Enabled lag sim of ${latency}ms`);
		transportManager.LatencySimulator.SetLatency(latency);
		transportManager.LatencySimulator.SetEnabled(true);
	}
}
